lib/hl7.ex

```diff
@@ -2,14 +2,201 @@ defmodule HL7 do
   @moduledoc """
   Utility functions to load HL7 files as local streams.
   """
-  require Logger
 
-  @buffer_size 32768
+  defstruct segments: []
 
+  @buffer_size 32768
   @type file_type_hl7 :: :mllp | :line | nil
 
+  @type hl7_map_data() :: %{
+          optional(non_neg_integer) => hl7_map_data() | String.t(),
+          e: non_neg_integer()
+        }
+  @type hl7_list_data() :: String.t() | [hl7_list_data]
+
+  @type segment() :: %{
+          0 => String.t(),
+          optional(pos_integer) => hl7_map_data() | String.t(),
+          e: non_neg_integer()
+        }
+
+  @type t() :: %__MODULE__{segments: [segment()]}
+
+  @type parsed_hl7_segments :: t() | [segment()]
+  @type parsed_hl7 :: t() | segment() | [segment()] | hl7_map_data()
+
+  alias HL7.Path
```

Comment:
Should we add the path sigil here so people can start to only use the `HL7` module as the public face of the library?

Reply:
Possibly! Will test with it.

Reply:
Done!